src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/FrameOfT.cs

```diff
@@ -28,15 +28,22 @@ public Frame(IHttpApplication<TContext> application, FrameContext frameContext)
         /// The resulting Task from this loop is preserved in a field which is used when the server needs
         /// to drain and close all currently active connections.
         /// </summary>
-        public override async Task RequestProcessingAsync()
+        public override async Task ProcessRequestsAsync()
         {
+            var dataRecieved = false;
+
             try
             {
                 while (!_requestProcessingStopping)
                 {
                     TimeoutControl.SetTimeout(_keepAliveTicks, TimeoutAction.CloseConnection);
 
-                    InitializeHeaders();
+                    if (dataRecieved)
```

Comment:
Yeah see, maybe just naming then, with `Reset()` being first thing rather than last thing. Something like `SetupRequest()` `PrepareRequest()`, `InitalizeRequest()` etc?